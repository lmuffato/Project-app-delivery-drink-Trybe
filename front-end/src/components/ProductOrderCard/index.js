import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import ProductOrderStatus from '../ProductOrderStatus';
import OrderCardContainer from './styles';

const testids = (id) => ({
  seller: {
    id: `seller_orders__element-order-id-${id}`,
    status: `seller_orders__element-delivery-status-${id}`,
    date: `seller_orders__element-order-date-${id}`,
    price: `seller_orders__element-card-price-${id}`,
    address: `seller_orders__element-card-address-${id}`,
  },
  customer: {
    id: `customer_orders__element-order-id-${id}`,
    status: `customer_orders__element-delivery-status-${id}`,
    date: `customer_orders__element-order-date-${id}`,
    price: `customer_orders__element-card-price-${id}`,
    address: '',
  },
});

function ProductOrderCard({
  address,
  date,
  price,
  orderId,
  status,
  user,
}) {
  const testid = testids(orderId)[user];
  const orderStatus = () => (
    <div className="order-info-basic-mcontainer">
      <ProductOrderStatus
        full
        status={ status }
        testid={ testid.status }
      />
    </div>
  );
  const info2 = () => (
    <div className="order-info-basic-mcontainer">
      <div className="order-info-badges">
        <div
          className="info-badge"
          data-testid={ testid.date }
        >
          {date}
        </div>
        <div
          className="info-badge"
          data-testid={ testid.price }
        >
          {`R$ ${price.toFixed(2).replace('.', ',')}`}
        </div>
      </div>
    </div>
  );
  const userAdress = () => (
    <div
      className="order-info-adress"
      data-testid={ testid.address }
    >
      {address}

    </div>
  );

  return (
    <Link to={ `${orderId}` }>
      <OrderCardContainer>
        <div className="order-id">
          <div>
            <p className="order-label">Pedido</p>
            <p
              className="order-product-id"
              data-testid={ testid.id }
            >
              {orderId}
            </p>
          </div>
        </div>
        <div className="order-info">
          <div className="order-info-basic">
            {orderStatus()}
            {info2()}
          </div>
          {address && userAdress()}
        </div>
      </OrderCardContainer>
    </Link>
  );
}

ProductOrderCard.propTypes = {
  orderId: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  status: PropTypes.oneOf(['pending', 'delivered', 'preparing']).isRequired,
  address: PropTypes.string,
  user: PropTypes.oneOf(['customer', 'seller']),
};

ProductOrderCard.defaultProps = {
  address: undefined,
  user: 'customer',
};

export default ProductOrderCard;
