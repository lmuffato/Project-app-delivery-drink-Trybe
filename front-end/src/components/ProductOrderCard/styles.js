import styled from 'styled-components';

const OrderCardContainer = styled.div`
  display: grid;
  grid-template-columns: 140px 1fr;
  max-width: 550px;
  background-color: ${({ theme }) => theme['light-dark']};
  box-sizing: border-box;
  transition: box-shadow 0.15s, width 0.15s;
  box-shadow: 0 0px 2px 0 ${({ theme }) => theme.shadow};

  &:hover {
    box-shadow: 0 5px 12px 0 ${({ theme }) => theme.shadow};
  }
  
  .order-id {
    width: 100%;
    height: 100%;
    display: flex;
    box-sizing: border-box;
    padding: 10px;
    align-items: center;
    justify-content: center;
    background-color: white;

    .order-label, .order-product-id { margin: 0; text-align: center; }
    .order-label { font-size: 18px; }
    .order-product-id { font-size: 28px; }
  }
  
  .order-info-basic-mcontainer { 
    padding: 5px;
    flex: 1;
  }

  .order-info {
    display: flex;
    flex-flow: column nowrap;
    .order-info-basic {
      flex: 1;
      display: flex;
      flex-flow: row wrap;
    }
    .order-info-adress {
      padding: 10px 5px 10px 5px;
      font-size: 15px;
      overflow-wrap: anywhere;
      text-align: end;
    }
  }
  .order-info-badges {
    display: flex;
    height: 100%;
    flex-flow: column nowrap;
    justify-content: space-between;
    .info-badge {
      background-color: white;
      box-sizing: border-box;
      text-align: center;
      border-radius: 6px;
      width: 100%;
      padding-block: 12px;
      font-size: 18px;
      font-weight: bold;
    }
  }
  @media only screen and (max-width: 500px) {
    grid-template-columns: none;
  } 
`;

export default OrderCardContainer;
