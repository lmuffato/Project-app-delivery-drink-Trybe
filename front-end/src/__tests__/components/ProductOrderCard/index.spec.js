import React from 'react';
import { render } from '@testing-library/react';
import ProductOrderCard from '../../../components/ProductOrderCard/index';

describe('Snapshot test', () => {
  it('renders correctly', () => {
    const JSON = render(
      <ProductOrderCard
        orderId="1"
        date="20012020"
        price={ 2 }
        status="delivered"
        address="Rua X"
        user="customer"
      />,
    );
    expect(JSON).toMatchSnapshot();
  });
});
