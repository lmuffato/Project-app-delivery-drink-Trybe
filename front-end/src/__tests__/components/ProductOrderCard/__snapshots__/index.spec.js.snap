// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsDKAQ fHwsMH"
      >
        <div
          class="order-id"
        >
          <div>
            <p
              class="order-label"
            >
              Pedido
            </p>
            <p
              class="order-product-id"
              data-testid="customer_orders__element-order-id-1"
            >
              1
            </p>
          </div>
        </div>
        <div
          class="order-info"
        >
          <div
            class="order-info-basic"
          >
            <div
              class="order-info-basic-mcontainer"
            >
              <div
                class="sc-bdvvtL aSatC"
                type="delivered"
              >
                <span
                  data-testid="customer_orders__element-delivery-status-1"
                >
                  ENTREGUE
                </span>
              </div>
            </div>
            <div
              class="order-info-basic-mcontainer"
            >
              <div
                class="order-info-badges"
              >
                <div
                  class="info-badge"
                  data-testid="customer_orders__element-order-date-1"
                >
                  20012020
                </div>
                <div
                  class="info-badge"
                  data-testid=""
                >
                  R$ 2.00
                </div>
              </div>
            </div>
          </div>
          <div
            class="order-info-adress"
            data-testid=""
          >
            Rua X
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsDKAQ fHwsMH"
    >
      <div
        class="order-id"
      >
        <div>
          <p
            class="order-label"
          >
            Pedido
          </p>
          <p
            class="order-product-id"
            data-testid="customer_orders__element-order-id-1"
          >
            1
          </p>
        </div>
      </div>
      <div
        class="order-info"
      >
        <div
          class="order-info-basic"
        >
          <div
            class="order-info-basic-mcontainer"
          >
            <div
              class="sc-bdvvtL aSatC"
              type="delivered"
            >
              <span
                data-testid="customer_orders__element-delivery-status-1"
              >
                ENTREGUE
              </span>
            </div>
          </div>
          <div
            class="order-info-basic-mcontainer"
          >
            <div
              class="order-info-badges"
            >
              <div
                class="info-badge"
                data-testid="customer_orders__element-order-date-1"
              >
                20012020
              </div>
              <div
                class="info-badge"
                data-testid=""
              >
                R$ 2.00
              </div>
            </div>
          </div>
        </div>
        <div
          class="order-info-adress"
          data-testid=""
        >
          Rua X
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
